docs/persistence.rst

```diff
@@ -0,0 +1,81 @@
+.. _persistence:
+
+Secure persistence with skops
+=============================
+
+.. warning::
+
+   This feature is very early in development, which means the API is
+   unstable and it is **not secure** at the moment. Therefore, use the same
+   caution as you would for ``pickle``: Don't load from sources that you
+   don't trust. In the future, more security will be added.
+
+Skops offers a way to save and load sklearn models without using :mod:`pickle`.
+The ``pickle`` module is not secure, but with skops, you can securely save and
+load sklearn models without using ``pickle``.
+
+Pickle is the standard serialization format for sklearn and for Python in
+general. One of its biggest advantages is it can be used for almost all Python
+code but this flexibility also makes it inherently insecure. This is because
+loading certain types of objects requires the ability to run arbitrary code,
+which can be misused for malicious purposes. For example, an attacker can use it
+to steal secrets from your machine or install a virus. As the Python docs say:
+
+    The pickle module is not secure. Only unpickle data you trust. It is
+    possible to construct malicious pickle data which will execute arbitrary
+    code during unpickling. Never unpickle data that could have come from an
+    untrusted source, or that could have been tampered with.
+
+In contrast to pickle, the :func:`skops.io.save` and :func:`skops.io.load`
+functions cannot be used to save arbitrary Python code, but they bypass
+pickle and are thus more secure.
+
+Usage
+-----
+
+The code snippet below illustrates how to use :func:`skops.io.save` and
+:func:`skops.io.load`:
+
+.. code:: python
+
+    from sklearn.linear_model import LogisticRegression
+    from skops.io import load, save
+
+    clf = LogisticRegression(random_state=0, solver="liblinear")
+    clf.fit(X_train, y_train)
+    save(clf, "my-logistic-regression.skops")
+    # ...
+    loaded = load("my-logistic-regression.skops")
+    loaded.predict(X_test)
+
+At the moment, we support the vast majority of sklearn estimators. This includes
+complex use cases such as :class:`sklearn.pipeline.Pipeline`,
+:class:`sklearn.model_selection.GridSearchCV`, classes using Cython code, such
+as :class:`sklearn.tree.DecisionTreeClassifier`, and more. If you discover an sklearn
+estimator that does not work, please open an issue on the skops `GitHub page
+<https://github.com/skops-dev/skops/issues>`_ and let us know.
+
+In contrast to pickle, skops cannot persist arbitrary Python code. This means
```

Comment:
We can probably add support while making it explicit that they are loading arbitrary code.

```suggestion
In contrast to pickle, at the moment skops cannot persist arbitrary Python code. This means
```

Reply:
there might be a way a way to support certain things by using `inspect.getsource`. I'm still not sure what it'd support, but we can have an explicit `arbitrary_code` type which we always warn on.

Reply:
How about we change the wording once we're there? :)

Reply:
Sure.